src/go/gcsrunner/fetch_config.go

```diff
@@ -109,31 +120,38 @@ func readBytes(opts FetchConfigOptions) ([]byte, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), opts.FetchGCSObjectTimeout)
 	defer cancel()
 
-	var client gcsReader
-	var err error
-	if opts.ServiceAccount == "" {
-		client, err = newDefaultCredsGCS(ctx)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		client, err = newGCS(ctx, opts.ServiceAccount)
-		if err != nil {
-			return nil, err
-		}
-	}
 	ebo := backoff.NewExponentialBackOff()
 	ebo.InitialInterval = opts.FetchGCSObjectInitialInterval
 	var out []byte
 	op := func() error {
 		if err := ctx.Err(); err != nil {
 			return backoff.Permanent(err)
 		}
+
+		var client gcsReader
+		var err error
+		if opts.ServiceAccount == "" {
+			client, err = newDefaultCredsGCS(ctx)
+			if err != nil {
+				glog.Errorf("error getting default creds GCS client (retrying): %v", err)
+				return err
+			}
+		} else {
+			client, err = newGCS(ctx, opts.ServiceAccount)
+			if err != nil {
+				glog.Errorf("error getting GCS client using service account (retrying): %v", err)
+				return err
+			}
+		}
```

Comment:
Won't this create a new client every time we retry? It might be preferred to move the `var client` outside the op, and that that it's non-nil before trying to create a new client each time.

Reply:
Added commit to do this